krnlmon_main.cc

```diff
@@ -0,0 +1,59 @@
+// Copyright 2022 Intel Corporation
```

Comment:
Implementation note: changed from C to C++ so we could use `absl::Notification` as our synchronization vehicle.

We might also consider using `std::thread` instead of pthread functions.